duplicate_generate.go

```diff
@@ -57,7 +57,7 @@ func main() {
 			continue
 		}
 
-		if name == "PrivateRR" || name == "OPT" {
+		if name == "PrivateRR" || name == "OPT" || name == "XPF" {
```

Comment:
This generator will work fine for this type. All it needs is to be updated to understand that a new ‘possibly A or AAAA’ field type should be compared as an IP address like `dns:"a"`.